Holovibes/sources/core/icompute.cc

```diff
@@ -169,6 +177,27 @@ void ICompute::update_spatial_transformation_parameters()
         CUDA_C_32F);                    // Computation type
 }
 
+void ICompute::allocate_moments_buffers()
+{
+    if (api::get_data_type() == RecordedDataType::MOMENTS)
+    {
+        auto frame_res = input_queue_.get_fd().get_frame_res();
+        auto batch_size = api::get_batch_size();
+
+        size_t size = frame_res * batch_size; // Batch size should be 1, but just in case.
+
+        moments_env_.moment0_buffer.resize(size);
+        moments_env_.moment1_buffer.resize(size);
+        moments_env_.moment2_buffer.resize(size);
+    }
+    else
+    {
+        moments_env_.moment0_buffer.reset(nullptr);
+        moments_env_.moment1_buffer.reset(nullptr);
+        moments_env_.moment2_buffer.reset(nullptr);
```

Comment:
Delete this ! This condition is done when the RecordedDataType is defferent from moment and it will clear the moments buffer.